src/lib.rs

```diff
@@ -47,6 +49,68 @@ impl SourceMapGenConfig for SourceMapConfig {
     }
 }
 
+#[derive(Default, Debug)]
+struct ContentTagVisitor {
+    occurrences: Vec<Occurrence>,
+}
+
+impl ContentTagVisitor {
+    fn add_occurrence(
+        &mut self,
+        span: &Span,
+        opening: &ContentTagStart,
+        contents: &ContentTagContent,
+        closing: &ContentTagEnd,
+    ) {
+        let occurrence = Occurrence {
+            range: span.into(),
+            content_range: contents.span.into(),
+            end_range: closing.span.into(),
+            start_range: opening.span.into(),
+            tag_name: "template".to_owned(),
+            kind: "template-tag".to_owned(),
```

Comment:
`kind: "template-tag"` is not doing anything useful here, since it's the only kind that's supported and that will likely ever be supported. In the earlier implementation it existed to distinguish between this and the hbs-backticks alternative proposal.

Instead, we should change this to have the value `"expression"` or `"class-member"` because that is highly-salient information that is already present in the AST. There were significant hacks required to work around *not* having that information in places like the eslint plugin, which is a big part of what motivated content-tag in the first place.

Reply:
Fixed in 05425a21e29bb4f8960be482940112251ca7cb52

Reply:
The AST doesn't currently differentiate it, so it's outside of the scope of the current PR, but we should eventually have a `declaration` kind too. The current AST just treat it as a top-level free-floating expression which isn't quite right. However, I want to refactor the AST to have a common `ContentTag` node, which `ContentTagExpression` and `ContentTagMember` can wrap, similar to how `Function` is the shared type for `FunctionExpression` and `Getter`, etc. That should make it easier to share code between the AST visitors.

Reply:
As for the use case for `declaration`: for example it would make it easier for the downstream tools to enforce that there can only be one per module, which isn't necessarily the case for _all_ use cases of content tag, but it is for the `<template>` specification.

Reply:
Visitor recursion handled in 51379ed9a2ca81afb0018ce2946c8c9674abc81c